src/main/java/org/mondemand/Context.java

```diff
@@ -35,18 +33,9 @@ public String getValue() {
     return value;
   }
 
-  public Context(String k, String v) throws MondemandException {
+  public Context(String k, String v) {
     key = k;
     value = v;
-    if(key == null) {
-      throw new MondemandException("key is null");
-    }
-    if(value == null || value.isEmpty()) {
-      throw new MondemandException("value is empty or null");
-    }
-    if(!isKeyValid(key)) {
-      throw new MondemandException("key is invalid: " + key);
-    }
```

Comment:
would removing this make it possible to get a bad context in if addContext were bypassed?

Reply:
what is a bad context though? only non-empty key/values should be ok?

Reply:
I can put the checks for `null` back and only drop the one for `isKeyValid()` if that is the right thing to do.

Reply:
yeah, I think null checks are enough.  you can remove the redundant null checks from addContext(), right?

Reply:
yeah what nick said

Reply:
removed null check from `addContext`, only kept the `null` checks in constructor of `Context` and removed the `isKeyValid()` from it.